src/common/transformations/src/transformations/common_optimizations/fuse_moe_expert.cpp

```diff
@@ -0,0 +1,340 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/fuse_moe_expert.hpp"
+
+#include <cstdint>
+#include <limits>
+
+#include "itt.hpp"
+#include "openvino/core/graph_util.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/util/shape_of_base.hpp"
+#include "openvino/opsets/opset1.hpp"
+#include "openvino/opsets/opset11.hpp"
+#include "openvino/opsets/opset12.hpp"
+#include "openvino/opsets/opset13.hpp"
+#include "openvino/opsets/opset15.hpp"
+#include "openvino/opsets/opset6.hpp"
+#include "openvino/opsets/opset8.hpp"
+#include "openvino/pass/pattern/matcher.hpp"
+#include "openvino/pass/pattern/op/optional.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "ov_ops/moe_expert.hpp"
+#include "ov_ops/type_relaxed.hpp"
+#include "transformations/utils/gen_pattern.hpp"
+#include "transformations/utils/utils.hpp"
+
+using namespace ov::gen_pattern;
+using namespace ov::pass;
+
+ov::pass::FuseMoeExpert::FuseMoeExpert() {
+    MATCHER_SCOPE(FuseMoeExpert);
+
+    auto expert_mask = makePattern(ov::Rank(3));
```

Comment:
@CuriousPanCake could you provide a short transition guide to our new symbolic infrastructure? does it cover all presented cases?
should we refine it now or handle it later?

Reply:
Is there guide suitable for transitioning?

Reply:
As agreement, let's follow with another PR, track in CVS-169302